iceoryx_posh/test/moduletests/test_roudi_memory_provider.cpp

```diff
@@ -302,3 +346,12 @@ TEST_F(MemoryProvider_Test, SegmentIdValueAfterDestructionIsUnset)
 
     EXPECT_THAT(sut.segmentId().has_value(), Eq(false));
 }
+
+TEST_F(MemoryProvider_Test, GetErrorString)
+{
+    for(int16_t i = 0; i < nTestCase;i++)
+    {
+        const char * result = MemoryProviderFailingCreation::getErrorStringMemoryProvider(m_testCombinationMemoryProviderError[i]);
```

Comment:
Can we not just call ``MemoryProvider::getErrorString`` directly here? For me it seems the only thing the test does is check the translation of enum values to strings. This is ok, but if those enum value or string array is not used elsewhere (did not fully check), they may be better placed in the test itself and not in the test fixture.

I think this test is not very meaningful in itself. What is needed to test failed memory creation and is it done in a sufficient manner elsewhere?

Reply:
Yes, indeed it is only testing the translation. But in order to improve the coverage, that's the only I found. We cannot just call the method because it is protected. That's why I implemented another method that's public and can be called in test.

Reply:
you can just add `using MemoryProvider::getErrorString;` in the public section of `MemoryProviderFailingCreation` to access it.

Reply:
I see. It is for coverage purposes. Now I understand the inheritance step, so feel free to ignore my comment regarding the necessity of this function.

This is somewhat unfortunate, because we are testing a protected function (i.e. implementation detail) instead of forcing the error by using the public API (this might be not easy in this case, depending on the error).

Ideally the error should be created by some use of public API functions, which in turn leads to the coverage as well. I do not know whether this is possible here.

The problem is, that we are increasing the coverage without actually testing whether the errors are actually under the conditions we expect. This gives a false sense of security/coverage.

@elBoberido We need to evaluate how it can be usefully tested (maybe not in this PR, I am not sure).

Reply:
Hi Matthias, we are testing all the input and checking whether the outputs are correct. So, on a level of unit test, I would say it is sufficient.

Reply:
@MatthiasKillat in this special case it's just a conversion method to get the string representation of the enum. It might even be better to implement the ostream operator for `MemoryProviderError`.

So this test doesn't try to trigger any error, it just checks if the string translation works.